### PR TITLE
Implementation of a simple Auth system

### PR DESCRIPTION
**Pull request by LuckyCyborg**
This pull-request introduce a simple but efficient **Auth** system, in a Laravel-esque style.

All the classes of that **Auth** system live in the namespace **\Auth\** and is implemented also a reference structure for User Authentication in the **\App\** namespace.

To note that additional **Route Filters** are added, to support this reference implementation and the properly configuration of **Helpers\Encrypter**, as in valid **ENCRYPT_KEY**, is required.

Being an **Users Management**, of course, a Database is required and in **scripts/nova_users.sql** you can find the associated MySQL dump for the **users** Table. 

Important: as Database API is used the new one, living in the namespace **\Database\**, then if you choose to use this Auth System, you should use that new Database API in the whole application and to not touch the **Helpers\Database** instances.

The reference implementation contains a class called **App\Controllers\Auth** which control the Users **login** and **logout**, as a base for your Users management.

The main **Auth System** classes are:

**Auth\Auth** - aliased as **\Auth**, which contains a simple manager of the current **Guard** instance. 
**Auth\Guard** - the Authentication Guard, which handle the User authentication.
**Auth\Model** - an Users base **Model**, built on top of **Database\Model** and configurable via a Config entry.

The offered services are:
**User Login** with **remembering** - the solution use **a strong encrypted Cookie** for remembering the authenticated user. I.e.
```php
// Prepare the Authentication credentials.
$credentials = array(
    'username' => Request::post('username'),
    'password' => Request::post('password')
);

// Prepare the 'remember' parameter.
if (Request::post('remember') == 'on') {
    $remember = true;
} else {
    $remember = false;
}

if(Auth::attempt($credentials, $remember)) {
    // User is authenticated there.
}
```

**User Logout** - using a simple command:
```php
Auth::logout();
```

**Authentication check**, via a boolean command:
```php
if(Auth::check()) {
    // User is authenticated there.
}
```
**Current User information** - the Auth have the ability to retrieve the authenticated User information, in a **stdClass** instance, fetched via **PDO::FETCH_OBJ**. No ORM Entities or data arrays are supported. i.e.
```php
if(Auth::check()) {
    // User is authenticated there, retrieve its Record.
    $user = Auth::user();

   // Do something cool there.
}
```

Finally, to note that this Auth System do **NOT** offer support for **User Registration** or **Credentials Recovery**, being all about a simple, yet secure and strong, **User Authentication**.

### CODE REVIEW

File: app/Config.example.php [hunk 1]
```diff
@@ -92,7 +92,7 @@
 define('SITETITLE', 'Nova V3.0');
 
 /**
- * Define a 32 bits key
+ * Define a 32 bits Encription Key.
```
Comment (tomvlk) [inline app/Config.example.php]: Typo

Comment (LuckyCyborg) [inline app/Config.example.php]: Thanks!

File: system/Auth/Guard.php [hunk 2]
```diff
@@ -0,0 +1,359 @@
+<?php
+/**
+ * Giuard - A simple Authentication Guard.
```
Comment (Kingorgg) [inline system/Auth/Guard.php]: Giuard -> Guard

Comment (LuckyCyborg) [inline system/Auth/Guard.php]: Thanks, will be fixed in next PR

**Comment by geomorillo** [comment]
Nice nice

**Comment by jimgwhit** [comment]
@LuckyCyborg can I get user info like  in laravel doc?
~~~

Accessing The Logged In User

Once a user is authenticated, you may access the User model / record:

$email = Auth::user()->email;
To retrieve the authenticated user's ID, you may use the id method:

$id = Auth::id();
To simply log a user into the application by their ID, use the loginUsingId method:

Auth::loginUsingId(1);
~~~

**Comment by LuckyCyborg** [comment]
@jimgwhit I choose to implement only the basic commands to, and there is no ORM, then what returns Auth::user() is an **stdClass** instance fetched via **PDO::FETCH_OBJ** from database and which contains all the User record information. Then, that will work:

```php
$email = Auth::user()->email;
```

But, no **Auth::loginUsingId()** for you. If there is a need for, could be implemented in future.

**Pull request merged**